### PR TITLE
Verify cmake build works from dist tarballs

### PR DESCRIPTION
**Pull request by rillian**
Add missing CMake build description files to the dist target and try to catch bugs like this in the the future.

### CODE REVIEW

**Comment by rillian** [comment]
this doesn't work on the older cmake travis provides. Working on a fix.

File: Makefile.am [hunk 1]
```diff
@@ -26,6 +26,17 @@ dist-hook:
 	    echo "OK"; \
 	  fi; \
 	done
+
+# Verify cmake works with the dist tarball.
+cmake_builddir = _build.cmake
+distcheck-hook:
+	$(RM) -rf $(cmake_builddir)
+	mkdir $(cmake_builddir)
```
Comment (tmatth) [inline Makefile.am]: instead of `rm` + `mkdir`, why not just `mkdir -p`?

Comment (rillian) [reply Makefile.am]: I wanted to make sure the directory was clean, without any cached cmake config from other builds.

**Comment by rillian** [comment]
Ok, merged. Thanks for the review!

**Pull request closed without merging**